dev/bus/tests/serial_unit.cpp

```diff
@@ -0,0 +1,83 @@
+#include <CppUTest/TestHarness.h>
+#include <CppUTest/CommandLineTestRunner.h>
+#include <CppUTestExt/MockSupport.h>
+
+#include "dev/serial.hpp"
+#include "mocks/platform_bus.hpp"
+#include "ecl/thread/semaphore.hpp"
+#include <thread>
+
+// Error code helper
+// TODO: move it to 'utils' headers and protect with check of
+// current test state (enabled or disabled)
+static SimpleString StringFrom(ecl::err err)
+{
+    return SimpleString{ecl::err_to_str(err)};
+}
+
+using serial_t = ecl::serial<platform_mock>;
+
+TEST_GROUP(serial)
```

Comment:
I propose to add following tests (or you can implement some kind of variations if you will):

 1. Multiple `recv_byte` with total count less than internal buffer size, provided that there is enough data in the internal buffer. I.e. `recv_byte` will not block. 
 1. Multiple `recv_byte` with total count less than internal buffer size, provided that there is not enough data in the internal buffer to satisfy all `recv_byte` calls.. `recv_byte` will block when buffer will be empty. Thread that emulates ISR will trigger event with new data after `recv_byte` will block (adding simple delay between thread start and event firing should do the trick).
 1. Multiple `recv_byte` with total count bigger than internal buffer size,  provided that the internal buffer is full. Case is almost the same as previous one, however it will also test wraparound.